drivers/sensor/hx711/hx711.c

```diff
@@ -0,0 +1,528 @@
+ *
+ */
+static int hx711_attr_set_gain(struct device *dev,
+							   const struct sensor_value *val)
+{
+	struct hx711_data *data = dev->driver_data;
+
+	switch (val->val1) {
+	case HX711_GAIN_128X:
+		data->gain = HX711_GAIN_128X;
+		break;
+	case HX711_GAIN_32X:
+		data->gain = HX711_GAIN_32X;
+		break;
+	case HX711_GAIN_64X:
+		data->gain = HX711_GAIN_64X;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return hx711_sample_fetch(dev, SENSOR_CHAN_WEIGHT);
+}
+
+
+/**
+ * @brief Set HX711 rate.
+ *
+ * @param dev Pointer to the hx711 device structure
+ * @param val sensor_value struct. Only val1 is used.
+ *            valid values are :
+ *               HX711_RATE_10HZ (default),
+ *               HX711_RATE_80HZ
+ *
+ * @retval 0 on success,
+ * @retval -EINVAL error if an invalid value is passed.
+ *
+ */
+static int hx711_attr_set_rate(struct hx711_data *data,
+							   const struct sensor_value *val)
+{
+	switch (val->val1) {
+	case HX711_RATE_10HZ:
+	case HX711_RATE_80HZ:
+		data->rate = val->val1;
+		return gpio_pin_set(data->rate_gpio, hx711_config.rate_pin, data->rate);
+	default:
+		return -EINVAL;
+	}
+
+
+}
+
+/**
+ * @brief Set HX711 offset.
+ *
+ * @param dev Pointer to the hx711 device structure
+ * @param offset sensor_value struct. Offset used to calculate the real weight.
+ *        Formula is :
+ *          weight = slope * (reading - offset)
+ *          Only val1 is used.
+ *
+ * @return void
+ *
+ */
+static void hx711_attr_set_offset(struct hx711_data *data,
+								  const struct sensor_value *offset)
+{
+	data->offset = offset->val1;
+}
+
+/**
+ * @brief Set HX711 offset.
+ *
+ * @param dev Pointer to the hx711 device structure
+ * @param slope sensor_value struct. Slope used to calculate the real weight.
+ *        Formula is :
+ *          weight = slope * (reading - offset)
+ *
+ * @return void
+ *
+ */
+static void hx711_attr_set_slope(struct hx711_data *data,
+								 const struct sensor_value *slope)
+{
+	data->slope.val1 = slope->val1;
+	data->slope.val2 = slope->val2;
+}
+
+/**
+ * @brief Set HX711 attributes.
+ *
+ * @param dev Pointer to the hx711 device structure
+ * @param chan Channel to set.
+ *             Supported channels :
+ *               SENSOR_CHAN_WEIGHT
+ *               SENSOR_CHAN_ALL
+ * @param attr Attribute to change.
+ *             Supported attributes :
+ *               SENSOR_ATTR_SAMPLING_FREQUENCY
+ *               SENSOR_ATTR_OFFSET
+ *               SENSOR_ATTR_SLOPE
+ *               SENSOR_ATTR_GAIN
+ * @param val   Value to set.
+ * @return void
+ *
+ */
+static int hx711_attr_set(struct device *dev,
+						  enum sensor_channel chan,
+						  enum sensor_attribute attr,
+						  const struct sensor_value *val)
+{
+	int ret = 0;
+	struct hx711_data *data = dev->driver_data;
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		if (hx711_attr_set_rate(data, val) == 0) {
+			LOG_DBG("Attribute RATE set to %d\n", data->rate);
+		}
+		return ret;
```

Comment:
Shouldn't you update ret based on the result of `hx711_attr_set_rate`?

Reply:
Yes I should.
Done.